### PR TITLE
Add wrap files for mpdecimal 2.5.0

### PR DESCRIPTION
**Pull request by dnicolodi**
I strongly suspect that this does not compile with MSVC, but I don't have a machine to test it right now.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+  'libmpdec++/decimal.cc',
+)
+
+libmpdecpp = library(
+  'mpdec++',
+  libmpdecpp_sources, mpdecimal_h,
+  version : library_version,
+  darwin_versions : [compat_version, library_version],
+  link_args : ['-Wl,-undefined,dynamic_lookup'],  
+  cpp_args : ['-fPIC'])
```
Comment (jpakkane) [inline meson.build]: This should  not use `-fPIC` but instead something like:

```
override_options: ['b_staticpic=true']
```

Comment (dnicolodi) [reply meson.build]: Thank you for the review. While directly passing `-fPIC` is probably not the right thing, I think `override_options: ['b_staticpic=true']` is not either. The original Makefile uses `-fPIC` only for the shared library.

Comment (jpakkane) [reply meson.build]: In that case you don't need to do anything. Meson does exactly that by default already.

Comment (dnicolodi) [reply meson.build]: Ah! Nice. Fixed

**Comment by jpakkane** [comment]
This fails to compile because it is missing the math library:

```
/tmp/tmpa04ahd24/bin/../src/mpdecimal-2.5.0/libmpdec/mpdecimal.c:4347: undefined reference to `ceil'
/usr/bin/ld: libmpdec.so.2.5.0.p/libmpdec_mpdecimal.c.o: in function `mpd_sizeinbase':
```

You might want to use the review tool from the Mesonwrap repo: https://github.com/mesonbuild/mesonwrap/

**Comment by dnicolodi** [comment]
Thank you for catching this. I tested it, but it compiles just fine without the math library on macOS. Should be fixed now.

**Comment by jpakkane** [comment]
The C++ one does not seem to link, missing libs?

```
/usr/bin/ld: libmpdec++.so.2.5.0.p/libmpdec++_decimal.cc.o: in function `(anonymous namespace)::LibraryInit::LibraryInit()':
/tmp/tmp1b2xh5kp/bin/../src/mpdecimal-2.5.0/libmpdec++/decimal.cc:42: undefined reference to `mpd_setminalloc'
/usr/bin/ld: libmpdec++.so.2.5.0.p/libmpdec++_decimal.cc.o: in function `decimal::IEEEContext(int)':
/tmp/tmp1b2xh5kp/bin/../src/mpdecimal-2.5.0/libmpdec++/decimal.cc:187: undefined reference to `mpd_ieee_context'
/usr/bin/ld: libmpdec++.so.2.5.0.p/libmpdec++_decimal.cc.o: in function `decimal::Context::repr[abi:cxx11]() const':
/tmp/tmp1b2xh5kp/bin/../src/mpdecimal-2.5.0/libmpdec++/decimal.cc:228: undefined reference to `mpd_round_string'
/usr/bin/ld: libmpdec++.so.2.5.0.p/libmpdec++_decimal.cc.o: in function `decimal::Decimal::exact(char const*, decimal::Context&)':
/tmp/tmp1b2xh5kp/bin/../src/mpdecimal-2.5.0/libmpdec++/decimal.cc:264: undefined reference to `mpd_qset_string_exact'
/usr/bin/ld: libmpdec++.so.2.5.0.p/libmpdec++_decimal.cc.o: in function `decimal::Decimal::ln10(long, decimal::Context&)':
```

**Comment by dnicolodi** [comment]
It links on macOS. It seems that these symbols are from `libmpdec` but I don't find anything in the Makefile that links the cpp library with the c library. Let me test this in a Linux VM.

**Comment by dnicolodi** [comment]
The last push should have fixed it. The original Makefile does nothing special and it works, however Meson uses `-Wl,--as-needed -Wl,--no-undefined` that of course is not happy with the undefined symbols. Linking explicitly with the C library solves the problem. On macOS the Makefile uses the `-undefined dynamic_lookup` linker option. Linking the C library to the C++ one removed the need for that too. I admit that linking is still obscure to me sometimes...

**Pull request merged**